src/packages/emmett/src/testing/features.ts

```diff
@@ -0,0 +1,90 @@
+/* eslint-disable @typescript-eslint/no-floating-promises */
+import assert from 'node:assert';
+import { randomUUID } from 'node:crypto';
+import { describe, it } from 'node:test';
+import type { EventStore } from '../eventStore';
+import {
+  evolve,
+  getInitialState,
+  type PricedProductItem,
+  type ShoppingCartEvent,
+} from './shoppingCart.domain';
+
+type TestOptions = {
+  getInitialIndex: () => bigint;
+};
+
+export type EventStoreFactory = () => Promise<{
+  eventStore: EventStore<bigint>;
+  tearDown?: () => Promise<void>;
+}>;
+
+export async function testAggregateStream(
```

Comment:
I still have to work on this. It only works for readily available Event Stores that don't need a setup or teardown phase.